fetch.md

```diff
@@ -1,11 +1,204 @@
+
+Like the main proposal, we could potentially provide an optional cache merging step in which resources stored
+in the ephemeral cache could be transferred to the HTTP cache for a more-permanent lifetime. This is not without
+its own difficulties and complexities as previously described.
+
+#### Bypassing the HTTP cache entirely
+
+Another approach to this problem would be to force-send all requests that were initiated even indirectly
+by a [prerendering browsing context][4] with the [cache mode][3] value of `"no-store"`. Fetching would behave
+as though HTTP cache were not present, and it would be impossible to contaminate any HTTP cache partitions observable
+to other pages.
+
+The main downside of this approach is that we lose the ability to use any of the resources fetched by
+the [prerendering browsing context][4] after activation. That is, we lose whatever benefits we could
+gain by having the ability to merge speculative <=> "typical" HTTP cache partitions, should we decide
+to go that route.
+
+On a similar note, this approach does not integrate with `<link rel=prefetch>`, as the whole point is
+to cache a top-level resource somewhere it can be usefully retrieved upon navigation. This means we're likely going to have
+to spec and implement the HTTP cache partitioning mechanics described above anyways, so it is sensible for
+other speculative navigation requests to piggy-back off of the same infrastructure.
+
+Another possible complexity is that if we spec prerendering in a way such that [prerendering browsing contexts][4]
+can be thrown away under implementation-defined constraints at any time, then we lose the ability to ever profit
+from the prerendering work we've done. No resources in the [prerendering browsing contexts][4] are preserved
+elsewhere, so it would be impossible to downgrade prerendering activation to something like prefetch activation.
+
+> TODO(domfarolino): Clarify that the above is only considering the downgrading of prerender *activations*. There is
+another kind of downgrade that we may want to consider, which is the scenario where under implementation-defined
+constraints, the UA opts not to create a [prerendering browsing context][4], but instead downgrade the whole prerender
+process to a simple prefetch. None of this is set in stone though.
+
+## Fetching With No Credentials
+
+Speculative navigation requests should in no way leak the user's identity to third-party origins. A consequence of
+this is that we must fetch all requests originating from a [prerendering browsing context][4] with no credentials.
+As with the HTTP cache partitioning restrictions, this must apply to all of the following:
+ - Top-level speculative resources
+ - Subresources under top-level speculative resources
+ - Nested browsing contexts and their subresources
+
+Concretely this can be done by overriding the [credentials mode][6] of all requests coming from a
+[prerendering browsing context][4] to the `"omit"` value.
```

Comment:
Given the separate network partition, what credentials are there likely to be to omit? Does this mean that multiple requests from the same prerendering browsing context (e.g. for subresources) cannot be correlated? Is that an explicit goal, or incidental?

I'd have expected it to be sufficient to start from an empty partition, so that the prefetching activity can't be correlated with any previous browsing activity.

Reply:
I believe cookie jar / credential partitioning is different than network partitioning. This is my understanding from:
 - https://fetch.spec.whatwg.org/#ref-for-concept-request-current-url%E2%91%A1%E2%91%A3, as well as
 - What I've heard about some potential future efforts to work on cookie jar partitioning in the same way we do HTTP cache & connection partitioning, and the fact that this is a separate effort

So basically I think that credentials are not a benefactor of all of the partitioning work we're doing today with the network partition key, but I'd like to sanity check with @domenic to be sure.

Reply:
I agree that they're separate efforts. However, we may also want to discuss a partitioned cookie jar approach? In particular if the main page does `Set-Cookie: ...`, maybe any same-origin `<img>` requests should get that cookie.

Reply:
Oh nevermind, I did include it in the form of a TODO in the credentials section. In that case, I think this is good to know since the task of elaborating on that alternate proposal is tracked there. I will work on further hashing that out asynchronously.